### PR TITLE
Improved exception handling

### LINKED ISSUE #13: Exception handling improvements
The error experience when not properly connected to a device could be improved.

Another case would be to explore errors for not having the app installed and making APDU specific errors come out okay.

### PR DESCRIPTION
**Pull request by mikeshultz**
Adds improved exception handling and resolution.

Fixes #13

### CODE REVIEW

File: ledgereth/comms.py [hunk 1]
```diff
@@ -81,31 +82,13 @@ def get_with_data(
...
-
-
 def dongle_send(dongle, command_string: str) -> bytes:
     """Send a command to the dongle"""
     hex_command = LedgerCommands.get(command_string)
     try:
         return dongle.exchange(hex_command)
     except CommException as err:
-        raise translate_exception(err)
+        raise LedgerError.transalate_comm_exception(err)
```
Comment (unparalleled-js) [inline ledgereth/comms.py:91]:
```suggestion
        raise LedgerError.transalate_comm_exception(err) from err
```

to preserve the exception stack?

File: ledgereth/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+class LedgerAppNotOpened(LedgerError):
+    message = "Expected Ledger Ethereum app not open"
+
+
+class LedgerCancel(LedgerError):
+    message = "Action cancelled by the user"
+
+
+class LedgerInvalid(LedgerError):
+    message = 'Invalid date sent to ledger or "blind signing" is not enabled'
```
Comment (unparalleled-js) [inline ledgereth/exceptions.py:87]:
```suggestion
    message = 'Invalid data sent to ledger or "blind signing" is not enabled'
```

**Review by unparalleled-js — COMMENTED** [review]
I added some thoughts! I haven't tried this out.
When we swap out our implementation in `ape-ledger` with this, I will test this out a lot more

File: ledgereth/comms.py (hunk 1 above)
Comment (mikeshultz) [reply ledgereth/comms.py:91]: I'm not familiar with this, can you point me towards the docs?  Not sure how I'd look that up.

File: tests/test_exceptions.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    LedgerInvalid,
+    LedgerLocked,
+    LedgerNotFound,
+)
+
+
+def test_comms_exception(yield_dongle):
+    with yield_dongle(exception=CommException("TEST", 0xFFFF, 0x00)) as dongle:
+        try:
+            dongle_send(dongle, "GET_CONFIGURATION")
```
[the remark this answers is not in the document]
Comment (mikeshultz) [reply tests/test_exceptions.py:18]: Cleaner, I like it

File: ledgereth/comms.py (hunk 1 above)
Comment (mikeshultz) [reply ledgereth/comms.py:91]: Found it.  https://docs.python.org/3/library/exceptions.html#exception-context

**Pull request merged**